nixos/modules/services/networking/v2raya.nix

```
@@ -6,11 +6,22 @@ with lib;
   options = {
     services.v2raya = {
       enable = options.mkEnableOption "the v2rayA service";
+
+      cliPackage = mkOption {
+        type = types.package;
+        default = pkgs.v2ray;
+        defaultText = literalExpression "pkgs.v2ray";
+        example = literalExpression "pkgs.xray";
+        description = ''
+          The package used for the cli binary.
+        '';
+        relatedPackages = [ "xray" "v2ray" ];
+      };
     };
   };
 
   config = mkIf config.services.v2raya.enable {
-    environment.systemPackages = [ pkgs.v2raya ];
+    environment.systemPackages = [ (pkgs.v2raya.override { v2ray = config.services.v2raya.cliPackage; }) ];
```

```suggestion
    environment.systemPackages = [ (cfg.package { v2ray = cfg.cliPackage; }) ];
```
Ditto for the other instance of the override.

---

Please push your changes, even if they're broken, so we can find the error.